### PR TITLE
switch ci from travis to actions

### PR DESCRIPTION
**Pull request by namusyaka**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: ${{ matrix.ruby }}
+        bundler-cache: true
+    - name: Install dependencies
+      run: sudo apt-get install -y pandoc nodejs pkg-config libxml2-dev libxslt-dev
+    - name: Run tests
+      run: ./.actions.sh
+
+# TODO(zzak, rkh): Add slack notification.
```
Comment (namusyaka) [inline .github/workflows/test.yml:38]: @zzak Probably we need to set up an integration with slack but I don't have a permission to generate a token on slack. Could you work on this?

**Comment by namusyaka** [comment]
Note that we should also update a status badge after merging this PR.
/cc @zzak @jkowens  @olleolleolle

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  test:
+    name: Test with Ruby-${{ matrix.ruby }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
+        # TODO: Add jruby if something like allow_failures will be implemented on Actions.
+        ruby: [2.3, 2.4, 2.5, 2.6, 2.7, '3.0', head]
+    env:
+      GITHUB_WORKSPACE: ${{ github.workspace }}
```
Comment (dentarg) [inline .github/workflows/test.yml:26]: I think this one is already set: https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables

Comment (namusyaka) [reply .github/workflows/test.yml]: Good catch, thanks.

**Review by olleolleolle — APPROVED** [review]
I think it is worth going forward, even without the Slack notification in place, yet.

**Review by zzak — APPROVED** [review]
Check #sinatra-core for the slack notification info :)

Thanks for working on this!

**Review by rkh — APPROVED** [review]

**Comment by zzak** [comment]
@namusyaka I've added the slack notification, but I think we may want to tweak it and possibly only run on `master`? Also how do you feel about changing the default branch?

**Comment by rkh** [comment]
+1 on changing the default branch to main.

File: .actions.sh [hunk 3]
```diff
@@ -5,14 +5,14 @@ echo "Running sinatra tests..."
 bundle exec rake
 
 echo "Running sinatra-contrib tests..."
-export BUILDIR=$TRAVIS_BUILD_DIR/sinatra-contrib
+export BUILDIR=$GITHUB_WORKSPACE/sinatra-contrib
```
Comment (swim86unifyd) [inline .actions.sh:8]: require 'sinatra'
    get '/frank-says' do
      'Put this in your pipe & smoke it!'
    end

**Review by swim86unifyd — COMMENTED** [review]
require 'sinatra'
    get '/frank-says' do
      'Put this in your pipe & smoke it!'
    end

**Pull request merged**